woocommerce/woocommerce-android
# Create display conditionals for the Feedback Request Card

Summary
==========
Fixes issue #2660 by making available a set of rules and conditionals with the following rules:

- The known installation date is more than 3 months ago
- The user has not given feedback for more than 6 months ago.

Changelog
==========
| Description | Commit |
| ------------- | ------------- |
| Create a extension property for Date objects where's possible to determine the amount of days from that Date object compared to today Date | b688474d14a1ebe570d038cb803337ecd21e9061 |
| Create SharedPreferences configuration to store the epoch timestamp as String within the `AppPrefs` object | 7241975deeae5056ff4d4381c098a68b90cb1050 |
| Create an infix function to work as a `greaterThan` operator that's capable to operate with nullable values | c895c280d1d3c428697c13b57cb4a53cb441cc10 |
| Add the conditionals to verify if the app should request feedback from the user | b69330fd395632ee6580ae01a757352bb8177262 |
| Insert the Feedback Due conditionals to the FeedbackRequestCard calls inside the `MyStoreFragment` and `DashboardFragment` | 5af895f236e864b418961636a868ad6ac17ba9f2 |
| Refactor the `MyStoreFragment` and `DashboardFragment` to better handle the visibility of the card whenever the device Local Date time is changed and modified | ad9a658cef3ea4b90810c2159043dff0e25092f2 |

Screenshots
==========
| Feedback Request Card Hidden  | Feedback Request Card Visible |
| ------------- | ------------- |
| ![screencap-2020-09-02T065522 019Z](https://user-images.githubusercontent.com/5920403/91943275-b4dea980-ecd2-11ea-904b-dda61dc9d914.png) | ![screencap-2020-09-02T065603 035Z](https://user-images.githubusercontent.com/5920403/91943278-b6a86d00-ecd2-11ea-8923-0a8df4da69c0.png) |

How to Test
==========
Scenario 1 - App installed for more than three months ago and no feedback was given
------------
1. Go to the device settings and change the system calendar to more than 90 days *BEFORE* of your local current date.
2. Make a fresh install of the Woo app from the scratch
3. Go back to the device settings and reactivate the automatic system calendar definition back to your local current date.
4. Go through the Login flow and verify if the Feedback card is visible
5. Verify if the card gets dismissed after any of both actions gets clicked
6. Verify if the card doesn't come back after reopening the app or through any kind of app interaction

Scenario 2 - App freshly installed and no feedback was given
------------
1. Make a fresh install of the Woo app from the scratch
2. Go through the Login flow and verify if the Feedback card is *NOT* visible
3. Verify if the card doesn't show up after reopening the app or through any kind of app interaction

Scenario 3 - Last Feedback was given six months ago
------------
1. Run through Scenario 1
2. Go to the device settings and change the system calendar to more than 180 days *AFTER* of your local current date.
3. Go back to the Woo app
4. Verify if the card is visible and clickable
5. Verify if the card gets dismissed after any of both actions gets clicked
6. Verify if the card doesn't come back after reopening the app or through any kind of app interaction

Update release notes:
